addon/components/ember-ckeditor.js

```diff
@@ -1,23 +1,130 @@
-/* globals CKEDITOR */
 import Ember from 'ember';
 import layout from '../templates/components/ember-ckeditor';
 
 export default Ember.Component.extend({
   layout: layout,
+  'on-change': null,
 
+  /**
+   * CK Editor Instance
+   */
   _editor: null,
-  'on-change': null,
+  /**
+   * Primary value. Bindable.
+   * Still have a bit of magic to ensure events are fired when they should be
+   */
+  value: null,
+  /**
+   * Ferries any bound-in values into the editor.
+   */
+  _handleValueChange: Ember.observer('value', function () {
+    if (this._editorChanging) {
+      return;
+    }
+    if (this._editor) {
+      this._editor.setData(this.get('value'));
+    }
+  }),
 
-  didInsertElement() {
-    let textarea = this.element.querySelector('.editor');
-    let editor = this._editor = CKEDITOR.replace(textarea);
-    editor.on('change', (e) => {
-      this.sendAction('on-change', e.editor.getData());
-    });
+  /**
+   * Skin that should be loaded.
+   * Not Bindable.
+   */
+  skin: 'moono',
+  /**
+   * The content tags that are permitted in the editor. Null/default will let the plugins configure.
+   * Note: If you remove the toolbar, you cannot use automatic content, and MUST specify it.
+   * Not Bindable.
+   */
+  allowedContent: null,
+  /**
+   * The content tags that are not permitted, even if the auto-config says they're available.
+   * Not Bindable.
+   */
+  disallowedContent: null,
+  /**
+   * ReadOnly mode
+   */
+  readOnly: false,
+  /**
+   * Build the config passed to ckeditor's init method
+   * Ideal for over-writing in sub-classes
+   * @returns {} ckeditor config
+   */
+  buildConfig: function () {
+    var config = {
+      skin: this.get('skin'),
+      disallowedContent: this.get('disallowedContent'),
+      readOnly: this.get('readOnly'),
+      allowedContent: this.get('allowedContent')
+    };
+
+    return config;
   },
+  _handleDidInsertElement: Ember.on('didInsertElement', function () {
```

Comment:
Mind sharing the rationale? The blogs and other opinions that I've read have been suggesting to move away from this for the last year. I haven't seen/heard any dissenting opinions, so I'd love to hear more about why you feel it's best to avoid it.